src/validators/problem.ts

```diff
@@ -28,7 +30,10 @@ export function problemValidator(method: string): ValidationChain[] {
                 body("source", "Invalid or missing 'source'").isString().exists(),
                 body("sourceLink", "Invalid or missing 'sourceLink'").isString().exists(),
                 body("problemSetIds", "Invalid or missing 'problemSetId'").isArray().exists(),
-                body("problemMetadata", "Invalid or missing 'problemMetadata' (optional)").optional()
+                body("problemSetIds", "Invalid or nonexistent 'problemSetId'").custom(validateProblemSetIds),
+                body("problemMetadata", "Invalid or missing fields in 'problemMetadata' (optional)")
+                    .optional()
```

Comment:
should our PUT request have optional parameters? I think the right way to do PUT is for the client to send the full object body, same as in a POST request